packages/lesswrong/components/votes/ThreeAxisEmojisVoteOnComment.tsx

```diff
@@ -0,0 +1,299 @@
+  voteProps: VotingProps<VoteableTypeClient>,
+  classes: ClassesType,
+}> = ({document, voteProps, classes}) => {
+  const currentUser = useCurrentUser();
+  const post = usePostsPageContext();
+  const {openDialog} = useDialog();
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [everOpened, setEverOpened] = useState(false);
+  const {captureEvent} = useTracking({
+    eventType: "emojiMenuClicked",
+    eventProps: {documentId: document._id, itemType: "comment"},
+  });
+
+  const onOpenMenu = useCallback((event: MouseEvent) => {
+    captureEvent("emojiMenuClicked", {open: true});
+    setAnchorEl(event.currentTarget as HTMLElement);
+    setEverOpened(true);
+  }, [captureEvent]);
+
+  const onCloseMenu = useCallback(() => {
+    captureEvent("emojiMenuClicked", {open: false});
+    setAnchorEl(null);
+  }, [captureEvent]);
+
+  const onSelectEmoji = useCallback((emojiOption: EmojiOption) => {
+    if (!currentUser) {
+      openDialog({
+        componentName: "LoginPopup",
+        componentProps: {},
+      });
+      return;
+    }
+
+    voteProps.vote({
+      document: voteProps.document,
+      voteType: voteProps.document.currentUserVote || null,
+      extendedVote: {
+        ...voteProps.document.currentUserExtendedVote,
+        [emojiOption.name]: !isEmojiSelected(voteProps, emojiOption),
+      },
+      currentUser,
+    });
+  }, [currentUser, openDialog, voteProps]);
+
+  const reactions = getCurrentReactions(voteProps);
+
+  const {EAEmojiPalette, LWTooltip} = Components;
+  return (
+    <>
+      {reactions.map(({emojiOption, score}) => {
+        const isSelected = isEmojiSelected(voteProps, emojiOption);
+        return (
+          <LWTooltip
+            key={emojiOption.name}
+            title={
+              <EmojiTooltipContent
+                currentUser={currentUser}
+                emojiOption={emojiOption}
+                isSelected={isSelected}
+                reactors={post?.commentEmojiReactors?.[document._id]}
+                classes={classes}
+              />
+            }
+            placement="top"
+            popperClassName={classes.tooltip}
+          >
+            <div
+              role="button"
+              onClick={() => onSelectEmoji(emojiOption)}
+              className={classNames(classes.button, {
+                [classes.buttonSelected]: isSelected,
+              })}
+            >
+              <div className={classes.emojiPreview}>{emojiOption.emoji}</div>
+              <div>{score}</div>
+            </div>
+          </LWTooltip>
+        );
+      })}
+      <div
+        role="button"
+        onClick={onOpenMenu}
+        className={classes.button}
+      >
+        <LWTooltip
+          title="Add reaction"
+          placement="top"
+          popperClassName={classes.tooltip}
+        >
+          <AddEmoji />
+        </LWTooltip>
+      </div>
+      <Menu
+        onClick={onCloseMenu}
+        open={Boolean(anchorEl)}
```

Comment:
I'm not a huge fan of this way of setting a menu open and closed, I know we use it in other places. Is there a way you could do it where the anchorEl is always set (as a `ref`) and we explicitly set whether the menu is open with a useState?